### PR TITLE
re-frame template project

### PR DESCRIPTION
**Pull request by kurtharriger**
Here is a fairly minimal re-frame demo project for those interested in using shadow-cljs with re-frame.

### CODE REVIEW

**Comment by thheller** [comment]
Thanks.

There is a template available already:
https://github.com/mhuebert/shadow-re-frame

Can you take out the cljsjs.* files since they are provided by https://github.com/thheller/shadow-cljsjs already?

File: re-frame/src/cljs/demo/core.cljs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            [demo.events :as events]
+            [demo.subs :as subs]
+            [demo.views :as views]
+            [demo.routes :as routes]
+            [demo.config :as config]))
+
+
+(defn dev-setup []
+  (when config/debug?
+    (enable-console-print!)
```
Comment (thheller) [inline re-frame/src/cljs/demo/core.cljs]: This is never required in `shadow-cljs` and done automatically by the compiler.

File: re-frame/test/cljs/demo/core_test.cljs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(ns demo.core-test
+  (:require [cljs.test :refer-macros [deftest is testing]]))
```
Comment (thheller) [inline re-frame/test/cljs/demo/core_test.cljs]: Just `:refer` please.

File: re-frame/shadow-cljs.edn [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                [day8.re-frame/trace "0.1.14"]
+                [binaryage/devtools "0.9.8"]
+                [cider/cider-nrepl "0.16.0"]
+                [secretary "1.2.3"]
+                [venantius/accountant "0.2.3"] ]
+
+ :source-paths ["dev"
+                "test/cljs"
+                "src/cljs"
+                "src/cljc"]
```
Comment (thheller) [inline re-frame/shadow-cljs.edn]: I'm absolutely no fan of this pattern and don't want to promote using it.

Please use a setup that does NOT separate code by extension.

```
:source-paths
["src/dev" "src/main" "src/test"]
```
This is what I usually do but `["dev", "src", "test"]` would be fine too.

**Comment by kurtharriger** [comment]
Thanks for feedback. I've updated the example.

File: re-frame/shadow-cljs.edn [hunk 4]
```diff
@@ -3,12 +3,10 @@
                 [binaryage/devtools "0.9.8"]
                 [cider/cider-nrepl "0.16.0"]
                 [secretary "1.2.3"]
-                [venantius/accountant "0.2.3"] ]
+                [venantius/accountant "0.2.3"]
+                [thheller/shadow-cljsjs "0.0.5"]]
```
Comment (thheller) [inline re-frame/shadow-cljs.edn]: No need to include it, shadow-cljs already depends on it and should always provide the latest version.

**Comment by thheller** [comment]
Thanks.

**Pull request merged**